### PR TITLE
Run with gunicorn

### PR DESCRIPTION
**Pull request by oyvindio**
Refactor startup mechanism to support running under a wsgi server as well as the Flask debug server

### CODE REVIEW

File: fiaas_skipper/__init__.py [hunk 1]
```diff
@@ -43,37 +33,39 @@ def _load_spec_config(spec_file):
...
+    return create_webapp(deployer, cluster), cfg
 
+app, cfg = create_app()
 
 if __name__ == "__main__":
-    main()
+    try:
+        app.run("0.0.0.0", cfg.port)
+    except BaseException:
+        log.exception("General failure! Inspect traceback and make the code better!")
```
Comment (mortenlj) [inline fiaas_skipper/__init__.py]: This `log` is undefined.

File: Dockerfile [hunk 2]
```diff
@@ -17,6 +17,7 @@ RUN pip wheel . --wheel-dir=/wheels/
 FROM common as production
 # Get rid of all build dependencies, install application using only pre-built binary wheels
 COPY --from=build /wheels/ /wheels/
-RUN pip install --no-index --find-links=/wheels/ --only-binary all /wheels/fiaas_skipper*.whl
+RUN pip install --no-index --find-links=/wheels/ --only-binary all /wheels/fiaas_skipper*.whl && \
+    pip install gunicorn==19.8.1 gevent=1.3.1
 EXPOSE 5000
-CMD ["skipper"]
+CMD ["/usr/local/bin/gunicorn", "--worker-class", "gevent" "--bind=0.0.0.0:5000", "--log-file=-", "fiaas_skipper:app"]
```
Comment (birgirst) [inline Dockerfile]: I'm guessing there is a default number of workers?

File: Dockerfile [hunk 3]
```diff
@@ -17,6 +17,7 @@ RUN pip wheel . --wheel-dir=/wheels/
 FROM common as production
 # Get rid of all build dependencies, install application using only pre-built binary wheels
 COPY --from=build /wheels/ /wheels/
-RUN pip install --no-index --find-links=/wheels/ --only-binary all /wheels/fiaas_skipper*.whl
+RUN pip install --no-index --find-links=/wheels/ --only-binary all /wheels/fiaas_skipper*.whl && \
+    pip install gunicorn==19.8.1 gevent==1.3.1
```
Comment (mortenlj) [inline Dockerfile]: You need to build wheels for gunicorn and gevent in the build stage above.

**Comment by oyvindio** [comment]
Closed in favor of #38 because of problems getting gunicorn to run the thing properly

**Pull request closed without merging**